content/token22-metadata.md

```diff
@@ -0,0 +1,1083 @@
+---
+title: Metadata and Metadata Pointer Extension
+objectives:
+- Explain how the metadata pointers and metadata extensions work on Token Extension Program Mints
+- Create an NFT with metadata embedded in the mint account itself
+- Create an NFT with the metadata pointer extension
+---
+
+# Summary
+- The `metadata pointer` extension associates a public key directly to a metadata account, whether it's stored internally or via an external metadata account like Metaplex. This setup simplifies the association of tokens with their metadata.
+- The `metadata` mint extension allows embedding of metadata directly into mint accounts through the Token Extension Program, typically utilized alongside a self-referencing metadata pointer. This facilitates embedding comprehensive token information at the minting stage.
+- These extensions enhance the interoperability of tokens across different applications and platforms by standardizing how metadata is associated and accessed.
+- Directly embedding or pointing to metadata can streamline transactions and interactions by reducing the need for additional lookups or external calls.
+
+# Overview
+
+The Token Extension Program streamlines metadata on Solana. Without the Token Extension Program, developers store metadata in metadata accounts using a metadata on-chain program; mainly `Metaplex`. However, this has some drawbacks. For example the mint account to which the metadata is "attached" has no awareness of the metadata account. To determine if an account has metadata, we have to PDA the mint and the `Metaplex` program together and query the network to see if a Metadata account exists. Additionally, to create and update this metadata you have to use a secondary program (i.e. `Metaplex`). These processes introduces vender lock and increased complexity. Token Extension Programs's Metadata extensions fix this by introducing two extensions:
```

Comment:
Very clear! One minor typo, 'vendor lock and' should be 'vendor lock in and' but otherwise very clear.